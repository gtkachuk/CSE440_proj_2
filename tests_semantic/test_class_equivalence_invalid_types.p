program testClassEquivalenceInvalidTypes;

class BB

BEGIN
   VAR row     : integer;
      yourBoat : boolean;
      streams  : ARRAY[0 .. 5] OF boolean;
END	       


class AA

BEGIN
   VAR row     : integer;
      yourBoat : boolean;
      streams  : ARRAY[0 .. 5] OF integer;
END		  


class testClassEquivalenceInvalidTypes

BEGIN
   
   VAR objectA	: AA;
       objectB	: BB;

FUNCTION testClassEquivalenceInvalidTypes;
BEGIN
   objectA := objectB
END

END
.
