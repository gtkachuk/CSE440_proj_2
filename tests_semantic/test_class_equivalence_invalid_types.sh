#!/bin/sh

. ../tests_lib.inc

test_diff_sem test_class_equivalence_invalid_types.p
