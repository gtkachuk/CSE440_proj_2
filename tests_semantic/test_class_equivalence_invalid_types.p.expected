ERROR:31:Class "bb" is not a base class of "aa" in the assignment
Errors detected. Exiting.
